core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -889,7 +1020,7 @@ object Chunk {
 /**
  * A chunk which has at least one element.
  */
-sealed trait NonEmptyChunk[+A] extends Chunk[A] {
+sealed trait NonEmptyChunk[+A] extends PolymorphicChunk[A] {
```

Comment:
Only if it doesn't extend `PolymorphicChunk`, which is pretty easy if we remove respecialization.